In [14]:
import numpy as np
import scipy
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

Suppose a random variable,
X, follows a Normal distribution.
Specifically, X∼N(55,σ= 2.3). Now, suppose we are going to roll a fair pentagonal trapezo-hedron die once (the die starts at a value of 1), the random variable, Y, represents the possible values we observe on the upper-most face of the die.

(a)  What values can be observed from the random variable
X
?

Infinite, while increasingly unlikely, there could be a data point 100 standard deviations away from the mean.

(b)  What values can be observed from the random variable
Y
?

In [3]:
import sys
y_data = np.arange(1,11)
np.savetxt(sys.stdout, y_data, newline=", ", fmt="%s")

1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 

(c)  What is the shape of the distribution for the random variable
X
?

Normalized

(d)  What is the shape of the distribution for the random variable
Y
?

Uniform

(e)  What is the value of the standard deviation for the random variable
X
?

2.3

(f)  What is the total area under the density curve for the random variable
X
?

1

(g)  50% of the values are above and below what number for the random variable
Y
?

In [4]:
y_data.mean()

5.5

(h)  What is the value of the mode for the random variable
X
?

55

(i)
Free response submission.
For each of the variables
X
and
Y
identify whether they are
discrete or continuous and explain why.

X: continuous, the values can take on the form of any real number.
Y: discrete, the values are limited to integers from 1 - 10.

2.
Soybeans.
Soybean yield in Iowa was historically low in 2012.  Yield of soybeans (bushels per
acre) in Iowa during 2012 can be described by a Normal distribution with a mean of 51 bushels
per acre and a standard deviation of 8 bushels per acre.  Use
the 68-95-99.7 Rule (Empirical
Rule)
to answer the following questions

In [5]:
mean = 51
stdev = 8

(a)  Use the template above to create a well labeled normal curve for the soybean yield (bushels
per acre) in Iowa during 2012.  On this graph numerically label the mean (iv), the center
68% (iii) and (v), the center 95% (ii) and (vi) and the center 99.7% (i) and (vii).  Report
answers  (i)  through  (vii.)   in  Blackboard.   You  should  also  use  this  picture  to  help  you
answer the next few questions.

In [6]:
stdev_n3 = i = mean - 3 * stdev
stdev_n2 = ii = mean - 2 * stdev
stdev_n1 = iii = mean - stdev
iv = mean
stdev_p1 = v = mean + stdev
stdev_p2 = vi = mean + 2 * stdev
stdev_p3 = vii = mean + 3 * stdev

print("i. -3" + "\u03C3" + ":", stdev_n3)
print("ii. -2" + "\u03C3" + ":", stdev_n2)
print("iii. -" + "\u03C3" + ":", stdev_n1)
print("iv. mean" + ":", mean)
print("v.", "\u03C3" + ":", stdev_p1)
print("vi. 2" + "\u03C3" + ":", stdev_p2)
print("vii. 3" + "\u03C3" + ":", stdev_p3)

i. -3σ: 27
ii. -2σ: 35
iii. -σ: 43
iv. mean: 51
v. σ: 59
vi. 2σ: 67
vii. 3σ: 75


(b)  The middle 95% of counties have a soybean yield
between
what two values?

In [7]:
print("{} and {}".format(stdev_n2, stdev_p2))

35 and 67


(c)  What is the value of the 16
th
percentile of soybean yield for counties in Iowa?

In [8]:
print(stdev_n1)

43


(d)  What proportion of counties have a soybean yield
between
35 and 75 bushels per acre?

In [9]:
# 75 -> mean + 3stdev (0.997)
# 35 -> mean - 2stdev (1 - 0.95 = 0.05)
0.997 - 0.05 

0.947

(e)  0.15% of counties have a soybean yield
less than or equal to what value?

In [10]:
print(stdev_n3)

27


(f)  What proportion of counties have a soybean yield of
at most
67 bushels per acre?

In [11]:
# 67 -> 2stdev -> 0.95
0.95

0.95

In [12]:
def add(a: int, b: int) -> int:
    '''Adds two numbers and return result'''
    return a + b

In [13]:
add

<function __main__.add>

3.

State Income Taxes.  All 50 U.S. states are given the ability to set their own state income tax rates, which is measured as a proportion of an individuals gross income.  In a recent study of state income taxes it was discovered that income tax rates for all 50 states follow a Normal
distribution with mean 5.0 and a standard deviation 0.75. 
State income tax rates are reported as a (proportion × 100).
Use this information to answer the following questions.

a. What is the probability a state is charging an income tax rate greater than 3.9?
Answer this question by completing parts (i) and (ii).

i. Provide the z-score corresponding to the income tax rate of 3.9.

In [122]:
from typing import List

def center(zscore):
    return 1 - scipy.stats.norm.sf(zscore) * 2

def right(zscore):
    return scipy.stats.norm.sf(zscore)

def left(zscore):
    return scipy.stats.norm.sf(zscore)

def no_center(zscore):
    return 1 - center(zscore)

def zscore_to_prob(zscore: float, area: List[str]=["center"]) -> float: 
    '''Given a zscore, return a decimal probability
    area default is center.
    area supports {center, left, right, no_center}'''
    
    areas = {
        "center": center,
        "left": left,
        "no_center": no_center,
        "right": right,
    }
    
    probs = (areas[str(x)](zscore) for x in area)
    return sum(probs)

def to_zscore(value, u, stdev):
    return (value - u) / stdev

In [114]:
zscore_to_prob(1, area=["center"]) - zscore_to_prob(2, ["left"])
zscore_to_prob

0.65993936018890664

In [125]:
mean = 5.0
stdev = 0.75
tax_39 = to_zscore(3.9, u=mean, stdev=stdev)
print("zscore for tax rate 3.9: ", tax_39)

zscore for tax rate 3.9:  -1.4666666666666668


ii. Based on your answer in (i), what is the probability a state is charging an income tax rate greater than 3.9?

In [127]:
zscore_to_prob(tax_39, ["right"])

0.92876662258601383

(b)  What is the probability a state is charging an income tax rate less than 5.3?
Answer this question by completing parts (i) and (ii).

i. Provide the z-score corresponding to the income tax rate of 5.3.

In [181]:
zscore = to_zscore(5.3, u=mean, stdev=stdev)
print("{:.2f}"
      .format(zscore))

0.40


ii. Based on your answer in (i), what is the probability a state is charging an income tax
rate less than 5.3?

In [138]:
print("Less than 5.3:", zscore_to_prob(zscore, ["left"]))

Less than 5.3: 0.34457825839


(c) Find Q1 (First Quartile). Answer this question by completing parts (i) and (ii).

i. Provide the z-score corresponding to Q1.

In [180]:
# Turns out scipy as cdf (zscore) and ppf (percentile to zscore)
import scipy.stats as st

q1 = st.norm.ppf(0.25)
print("Q1: {}"
      .format(q1))

Q1: -0.6744897501960817


ii. Based on your answer in (i), provide the value of Q1.

In [146]:
def zval(zscore, u, stdev):
    return u + stdev * zscore

In [179]:
print("Q1 value: {}"
      .format(zval(q1, u=mean, stdev=stdev)))

Q1 value: 4.494132687352939


(d) Find Q3 (Third Quartile).  Answer this question by completing parts (i) and (ii).

i. Provide the z-score corresponding to Q3.

In [178]:
q3z = st.norm.ppf(0.75)
print("q3 zscore {}"
      .format(q3z))

q3 zscore 0.6744897501960817


ii. Based on your answer in (i), provide the value of Q3.

In [177]:
print("Q3 value: {}"
      .format(zval(q3z, u=mean, stdev=stdev)))

Q3 value: 5.505867312647061


(e)  What is the value of the IQR for the distribution of income tax rates for all 50 states?

In [154]:
q3val = zval(st.norm.ppf(0.75), u=mean, stdev=stdev)
q1val = zval(st.norm.ppf(0.25), u=mean, stdev=stdev)

print(q3val - q1val)

1.01173462529


(f)  What is the income tax rate such that only 12.1% of all other states charge a higher income tax rate?  Answer this question by completing parts (i) and (ii).

i.  Provide the z-score corresponding to the above statement.

In [161]:
zscore = st.norm.ppf(1 - 0.121)
print(zscore)

1.1700024075


ii.  Based on your answer in (i), what is the income tax rate such that only 12.1% of all
other states charge a higher income tax rate?

In [162]:
tax_rate = zval(zscore, u=mean, stdev=stdev)
print(tax_rate)

5.87750180563


iii.  What percentile does the income tax rate obtained in part (ii) correspond to?  (Enter the value as a whole number, not a proportion.)

In [176]:
print("{}"
      .format(int(round(st.norm.cdf(zscore) * 100))))

88


(g)  What proportion of states charge an income tax rate within 1.25 standard deviations from
the mean?  Answer this question by completing parts (i) through (iii).

i.  Provide the SMALLER z-score corresponding to the above statement.

-1.25

ii.  Provide the LARGER z-score corresponding to the above statement.

1.25

iii.  Based on your answers in (i) and (ii), what proportion of states charge an income tax rate within 1.25 standard deviations from the mean?

In [184]:
print(zscore_to_prob(1.25, area=["center"]))

0.788700452666


In [187]:
# Right filled - left filled zscore = center
st.norm.cdf(1.25) - st.norm.cdf(-1.25)

0.7887004526662893

(h)
Free response submission.
Which state is more unusual, Nebraska with an income tax rate of 6.84 or Colorado with an income tax rate of 4.63?  Explain your answer.

In [204]:
nebraska = 6.84
colorado = 4.63

def val_to_zscore(val, u, stdev):
    return (val - u) / stdev

zscore_nebraska = val_to_zscore(nebraska, u=mean, stdev=stdev)
zscore_colorado = val_to_zscore(colorado, u=mean, stdev=stdev)

print("Nebraska: {}".format(zscore_nebraska))
print("Colorado: {}".format(zscore_colorado))

print("{} is more unusual due to a higher absolute zscore!"
      .format("Colorado"
              if zscore_colorado > zscore_nebraska
              else "Nebraska"))

Nebraska: 2.453333333333333
Colorado: -0.49333333333333346
Nebraska is more unusual due to a higher absolute zscore!


(i)
Free response submission.
Let’s say you live in a state with a relatively average income
tax rate of 4.67.  Provide an interpretation of your state’s income tax rate, i.e.  what can you say about the percentage of all other states that have a higher or lower income tax rate than yours?

In [222]:
zscore_my_state = val_to_zscore(4.67, u=mean, stdev=stdev)
percentage_below_my_state = st.norm.cdf(zscore_my_state) * 100
percentage_above_my_state = 100 - percentage_below_my_state

print("{:.3f}% states have a lower tax rate than I."
     .format(percentage_below_my_state),
     "\n{:.3f}% states have a higher tax rate than I."
     .format(percentage_above_my_state))

32.997% states have a lower tax rate than I. 
67.003% states have a higher tax rate than I.


(j)
Free  response  submission.
Would  the  mean  or  median  of  the  distribution  for  state
income tax rates be more useful to the IRS? Explain your answer.

The mean. The purpose of collecting taxes is to fund government activity and services. With no indication that the data is skewed or has any outliers either will work for estimating total income tax collected for the nation.

## 4. Iowa Unemployment.

A professor at Iowa State has a dataset from an unknown year con-
sisting of the unemployment rates for all 99 counties in the state of Iowa.  The professor asks her
students to conduct a statistical analysis of the data and reiterates that they may use any approach they deem necessary.  One of the students recalls that the Normal distribution is a widely
used density within the world of statistics and decides to conduct the analysis assuming all 99 counties unemployment rates follow a Normal distribution with mean μ and standard deviation σ. Use the data found in
`Unemployment.jmp`
to answer the following questions regarding the
students analytic approach.  Unemployment rates are reported as
a (proportion × 100).

In [225]:
df_unemployment = pd.read_csv("Unemployment.csv")
df_unemployment.describe()

Unemployment rate
count          99.000000
mean            3.958788
std             2.554365
min             0.220000
25%             2.050000
50%             3.500000
75%             5.550000
max            12.250000

(a)  Using the JMP file provided, what are the values of
μ and σ for this Normal model?

i.  Record the value for μ.  (Round your answer to 2 decimal places)

In [231]:
mean = df_unemployment.mean()[0]
print("mean: {:.2f}"
     .format(mean))

mean: 3.96


ii.  Record the value for σ.  (Round your answer to 2 decimal places)

In [234]:
stdev = df_unemployment.std()[0]
print("stdev: {:.2f}"
     .format(stdev))

stdev: 2.55


(b)  Assume the unemployment rates take on a Normal distribution with mean 4.0 and standard
deviation 2.6.  Find the probability a county has an unemployment rate greater than 8.108.

(c)  Again, assume the unemployment rates take on a Normal distribution with mean 4.0 and
standard deviation 2.6. What is the unemployment rate such that only 5.71% of all counties
have an unemployment rate smaller than that value?

(d)
Free response submission.
Use JMP to create a histogram and boxplot of the unem-
ployment rates (you do not need to upload the histogram or boxplot).  Based on the JMP
output,  describe  the  overall  shape  of  the  distribution  of  unemployment  rates  for  the  99
counties in Iowa.

(e)
Free response submission.
The lowest attainable unemployment rate is considered to
be the ”natural rate of unemployment.” This ”natural rate of unemployment” is required
to  be > 0.   Recall  that  the  Normal  distribution  can  take  on  any  possible  value.   Using
your answer to parts (4c) and (4d) give two reasons why it might not be okay using this
student’s statistical approach of the unemployment rates for all counties in Iowa.